akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -182,6 +182,9 @@ import akka.util.OptionVal
 
   private def withSafelyAdapted[U, V](adapt: () => U)(body: U => V): Unit = {
     Try(adapt()) match {
+      case Success(null) =>
+        ctx.log.warn(
+          "Adapter function returned null which is not valid as an actor message, ignoring. This can happen for example when using pipeToSelf and returning null from the adapt function. Null value is ignored and not passed on to actor.")
```

Comment:
Clarify log message by using "Message adapter function". We have so many different adapters. Also change tests that are looking for this.

Reply:
WDYT about this vs throwing/crashing the actor?

Reply:
Throwing from the adapter function is not nice because that will not involve supervision, and will unconditionally stop the actor. I think logging and ignoring is good.

Reply:
Oh, I didn't think it worked like that. 

Then we shouldn't do that in the adapt lambda in samples for ask/pipeToSelf for the failure case either. That's a shame because it's would be pretty neat to not have to make a failure message and instead throw and trigger supervision.

Reply:
Yes, but that is a very difficult problem to solve. We are not "inside" the behavior stack when we call the adapt function.

Reply:
Created an issue to discuss if we can figure out how to do that: https://github.com/akka/akka/issues/28592